service/storage_service.cc

```diff
@@ -2151,20 +2139,37 @@ future<> storage_service::raft_replace(raft::server& raft_server, raft::server_i
 
     // add myself to topology with request to replace
     while (!_topology_state_machine._topology.contains(raft_server.id())) {
-        slogger.info("raft topology: adding myself to topology for replace: {} replacing {}, ignored nodes: {}", raft_server.id(), replaced_id, ignored_ids);
         auto guard = co_await _group0->client().start_operation(&_abort_source);
 
         auto it = _topology_state_machine._topology.normal_nodes.find(replaced_id);
         if (it == _topology_state_machine._topology.normal_nodes.end()) {
             throw std::runtime_error(::format("Cannot replace node {}/{} because it is not in the 'normal' state", replaced_ip, replaced_id));
         }
 
-        for (const auto& ignored_id : ignored_ids) {
-            if (!_topology_state_machine._topology.find(ignored_id)) {
-                throw std::runtime_error(::format("ignored node {} is not found in the cluster", ignored_id));
+        std::unordered_set<raft::server_id> ignored_ids;
+        for (const sstring& n : ignore_nodes_strs) {
+            std::optional<raft::server_id> ignored_id;
+            try {
+                ignored_id = raft::server_id{utils::UUID(n)};
+            } catch (const std::exception&) {
```

Comment:
Can we delegate this parsing to `host_id_or_endpoint` constructor?
```cpp
struct host_id_or_endpoint {
    host_id id;
    gms::inet_address endpoint;

    enum class param_type {
        host_id,
        endpoint,
        auto_detect
    };

    host_id_or_endpoint(const sstring& s, param_type restrict = param_type::auto_detect);
```

Reply:
Also, is there a way to reduce some of the code duplication between here and `raft_removenode`?

Reply:
I made both changes.